storage/resource_mount_test.go

```diff
@@ -0,0 +1,1414 @@
+	assert.Equal(t, "", d.Get("source"))
+}
+
+func TestResourceAzureBlobMountCreateGeneric_Error_NoResourceID(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/clusters/get?cluster_id=b",
+				Response: compute.ClusterInfo{
+					State: compute.ClusterStateRunning,
+				},
+			},
+		},
+		Resource: ResourceDatabricksMount(),
+		CommandMock: func(commandStr string) common.CommandResults {
+			return common.CommandResults{
+				ResultType: "error",
+				Summary:    "Some error",
+			}
+		},
+		State: map[string]interface{}{
+			"cluster_id": "b",
+			"name":       "e",
+			"wasb": []interface{}{map[string]interface{}{
+				"auth_type":          "ACCESS_KEY",
+				"directory":          "/d",
+				"token_secret_key":   "g",
+				"token_secret_scope": "h",
+			}}},
+		Create: true,
+	}.Apply(t)
+	require.EqualError(t, err, "container_name or storage_account_name are empty, and resource_id or uri aren't specified")
+	assert.Equal(t, "", d.Get("source"))
+}
+
+func TestResourceAzureBlobMountGeneric_Read(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/clusters/get?cluster_id=b",
+				Response: compute.ClusterInfo{
+					State: compute.ClusterStateRunning,
+				},
+			},
+		},
+		Resource: ResourceDatabricksMount(),
+		CommandMock: func(commandStr string) common.CommandResults {
+			trunc := internal.TrimLeadingWhitespace(commandStr)
+			t.Logf("Received command:\n%s", trunc)
+			assert.Contains(t, trunc, "dbutils.fs.mounts()")
+			assert.Contains(t, trunc, `mount.mountPoint == "/mnt/e"`)
+			return common.CommandResults{
+				ResultType: "text",
+				Data:       "wasbs://c@f.blob.core.windows.net/d",
+			}
+		},
+		State: map[string]interface{}{
+			"cluster_id": "b",
+			"name":       "e",
+			"wasb": []interface{}{map[string]interface{}{
+				"auth_type":            "ACCESS_KEY",
+				"container_name":       "c",
+				"directory":            "/d",
+				"storage_account_name": "f",
+				"token_secret_key":     "g",
+				"token_secret_scope":   "h",
+			}},
+		},
+		ID:   "e",
+		Read: true,
+	}.Apply(t)
+	require.NoError(t, err)
+	assert.Equal(t, "e", d.Id())
+	assert.Equal(t, "wasbs://c@f.blob.core.windows.net/d", d.Get("source"))
+}
+
+func TestResourceAzureBlobMountGenericRead_NotFound(t *testing.T) {
+	qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.0/clusters/get?cluster_id=b",
+				Response: compute.ClusterInfo{
+					State: compute.ClusterStateRunning,
+				},
+			},
+		},
+		Resource: ResourceDatabricksMount(),
+		CommandMock: func(commandStr string) common.CommandResults {
+			trunc := internal.TrimLeadingWhitespace(commandStr)
+			t.Logf("Received command:\n%s", trunc)
+			return common.CommandResults{
+				ResultType: "error",
+				Summary:    "Mount not found",
+			}
+		},
+		State: map[string]interface{}{
```

Comment:
you can also supply `HCL` parameter - easier with type casting these horrible raw types and more readable